state-chain/runtime/src/lib.rs

```diff
@@ -1408,6 +1408,25 @@ impl_runtime_apis! {
 			all_prewitnessed_swaps
 		}
 
+		fn cf_scheduled_swaps(from: Asset, to: Asset) -> Vec<(Swap, BlockNumber)> {
+
+			let mut first_block = Swapping::first_unprocessed_block();
+			let last_block = System::block_number() + pallet_cf_swapping::SWAP_DELAY_BLOCKS;
+
+			let mut swaps = vec![];
+
+			while first_block <= last_block {
+				let swaps_selected_from_block = Swapping::swap_queue(first_block).into_iter()
+					.filter(|swap| { (swap.from == from && swap.to == to) || (swap.from == to && swap.to == from)})
+					.map(|swap| (swap, first_block));
+
+				swaps.extend(swaps_selected_from_block);
+				first_block += 1u32;
+			}
```

Comment:
Can remove the muts and increment: 
```rust
(first_block..=last_block).into_iter().map(|block| {
				Swapping::swap_queue(block).into_iter()
					.filter(|swap| { (swap.from == from && swap.to == to) || (swap.from == to && swap.to == from)})
					.map(move |swap| (swap, block))
			}).flatten().collect()
```

Reply:
Indeed. I had assumed the block number was generic here, and didn't implement Step.